src/main/scala/is/hail/annotations/ordering/VariantUnsafeOrdering.scala

```diff
@@ -0,0 +1,54 @@
+package is.hail.annotations.ordering
+
+import is.hail.variant._
+import is.hail.asm4s._
+import is.hail.annotations._
+import is.hail.expr._
+import is.hail.utils._
+
+class VariantUnsafeOrdering(t: TVariant, missingGreatest: Boolean) extends CodifiedUnsafeOrdering {
+  private val fundamentalComparators = t.representation.fields.map(_.typ.unsafeOrdering(missingGreatest)).toArray
+  private val repr = t.representation.fundamentalType
+
+  def compare(r1: MemoryBuffer, o1: Long, r2: MemoryBuffer, o2: Long): Int = {
+    val cOff1 = repr.loadField(r1, o1, 0)
+    val cOff2 = repr.loadField(r2, o2, 0)
+
+    val contig1 = TString.loadString(r1, cOff1)
+    val contig2 = TString.loadString(r2, cOff2)
+
+    val c = Contig.compare(contig1, contig2)
+    if (c != 0)
+      return c
+
+    var i = 1
```

Comment:
In terms of efficiency of generated byte code, would it make sense to compare the starts explicitly since they're very likely to disambiguate if contig did not already? Or is the difference at Scala compile time anyway?

Reply:
AFAIK, the only reason cseed unrolled the contig was because the type of `contig` in `TVariant.representation` is `TString` and we want to specifically use `Contig.compare` rather than the default string comparison.